File: app/src/main/java/com/sd/demo/visible_level/appview/BaseAppView.java
package com.sd.demo.visible_level.appview;

import android.content.Context;
import android.util.AttributeSet;
import android.util.Log;
import android.widget.FrameLayout;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.sd.demo.visible_level.MainActivity;
import com.sd.lib.vlevel.FVisibleLevelItem;

public class BaseAppView extends FrameLayout implements FVisibleLevelItem.VisibilityCallback
{
    public BaseAppView(@NonNull Context context)
    {
        super(context);
    }

    public BaseAppView(@NonNull Context context, @Nullable AttributeSet attrs)
    {
        super(context, attrs);
    }

    @Override
    public void onLevelItemVisibilityChanged(boolean visible, FVisibleLevelItem item)
    {
        Log.i(MainActivity.TAG, "AppView onLevelItemVisibilityChanged visible:" + visible
                + " item:" + item
                + " view:" + this);
    }
}

File: app/src/main/java/com/sd/demo/visible_level/appview/BaseAppView.kt
package com.sd.demo.visible_level.appview

import android.content.Context
import android.util.AttributeSet
import android.widget.FrameLayout
import com.sd.demo.visible_level.logMsg
import com.sd.lib.vlevel.FVisibleLevelItem

open class BaseAppView(
    context: Context,
    attrs: AttributeSet?,
) : FrameLayout(context, attrs), FVisibleLevelItem.Callback {

    override fun onLevelItemVisibilityChanged(item: FVisibleLevelItem) {
        logMsg {
            "${javaClass.simpleName} onLevelItemVisibilityChanged ${item.name} -> ${item.isVisible}"
        }
    }
}